aicsimageio/readers/tiff_reader.py

```diff
@@ -42,6 +75,156 @@ def _is_this_type(buffer: io.BufferedIOBase) -> bool:
                     return False
             return True
 
+    @staticmethod
+    def _imread(
+        img: Path,
+        scene: int,
+        page: int
+    ) -> np.ndarray:
+        # Load Tiff
+        with TiffFile(img) as tiff:
+            # Get proper scene
+            scene = tiff.series[scene]
+
+            # Get proper page
+            page = scene.pages[page]
+
+            # Return numpy
+            return page.asarray()
+
+    @property
+    def dask_data(self) -> da.core.Array:
+        """
+        Read a TIFF image file as a delayed dask array where each chunk of the constructed array is a delayed YX plane.
+
+        Returns
+        -------
+        img: dask.array.core.Array
+            The constructed delayed YX plane dask array.
+        """
+        if self._dask_data is None:
+            # Load Tiff
+            with TiffFile(self._file) as tiff:
+                # Check each scene has the same shape
+                # If scene shape checking fails, use the specified scene and update operating shape
+                scenes = tiff.series
+                operating_shape = scenes[0].shape
+                for scene in scenes:
+                    if scene.shape != operating_shape:
+                        operating_shape = scenes[self.specific_s_index].shape
+                        scenes = [scenes[self.specific_s_index]]
```

Comment:
Here you're modifying an iteration variable within the iteration loop. This is potentially dangerous, I'd suggest a `break` immediately after the log below to ensure no erroneous behavior

Reply:
Really good catch.